libs/application/templates/estate/src/fields/EstateMembersRepeater/AdditionalEstateMember.tsx

```diff
@@ -0,0 +1,217 @@
+                format="######-####"
+                required
+                backgroundColor="blue"
+                loading={queryLoading}
+                error={queryError ? formatMessage('error') : undefined}
+              />
+            </GridColumn>
+            <GridColumn span={['1/1', '1/2']} paddingBottom={2}>
+              <InputController
+                key={nameField}
+                id={nameField}
+                name={nameField}
+                defaultValue={field.name}
+                label={formatMessage(m.inheritanceNameLabel)}
+                error={error?.name ?? undefined}
+                readOnly
+              />
+            </GridColumn>
+            <GridColumn span={['1/1', '1/2']} paddingBottom={2}>
+              <SelectController
+                key={relationField}
+                id={relationField}
+                name={relationField}
+                label={formatMessage(m.inheritanceRelationLabel)}
+                defaultValue={field.relation}
+                options={relationOptions}
+                backgroundColor="blue"
+                error={error?.relation ?? undefined}
+              />
+            </GridColumn>
+          </>
+        ) : (
+          <>
+            <GridColumn span={['1/1', '1/2']} paddingBottom={2}>
+              <InputController
+                key={nameField}
+                id={nameField}
+                name={nameField}
+                backgroundColor="blue"
+                defaultValue={field.name}
+                error={error?.name ?? undefined}
+                label={formatMessage(m.inheritanceNameLabel)}
+              />
+            </GridColumn>
+            <GridColumn span={['1/1', '1/2']} paddingBottom={2}>
+              <DatePickerController
+                label={formatMessage(m.inheritanceDayOfBirthLabel)}
+                placeholder={formatMessage(m.inheritanceDayOfBirthLabel)}
+                id={dateOfBirthField}
+                key={dateOfBirthField}
+                name={dateOfBirthField}
+                locale="is"
+                maxDate={new Date()}
+                backgroundColor="white"
+                onChange={(d) => {
+                  setValue(dateOfBirthField, d)
+                }}
+                error={error?.dateOfBirth ?? undefined}
+              />
+            </GridColumn>
+            <GridColumn span={['1/1', '1/2']} paddingBottom={2}>
+              <SelectController
+                key={relationField}
+                id={relationField}
+                name={relationField}
+                label={formatMessage(m.inheritanceRelationLabel)}
+                defaultValue={field.relation}
+                options={relationOptions}
+                error={error?.relation ?? undefined}
+                backgroundColor="blue"
+              />
+            </GridColumn>
+          </>
+        )}
+        <GridColumn span="1/1" paddingBottom={2}>
+          <CheckboxController
```

Comment:
Could this component be broken up so that this custom component would only be the input field that uses the nameField with a condition `foreignCitizenship[0] !== YES` 
Where you would then make 2 conditionally rendered multifields with all these inputs as the standard BuildTextFormField and one custom component with the the custom logic with the nameField query?
Just for the sake of avoiding unnecessary complex custom components?
I might be missing something here so feel free to point it out if so 😄

Reply:
hmmm, sounds great! is it ok with you if i'll do this as a followup PR?